pkg/controllers/cspc-controller/status.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cspccontroller
+
+import (
+	cstor "github.com/openebs/api/pkg/apis/cstor/v1"
+	"github.com/openebs/api/pkg/apis/types"
+	"github.com/openebs/cstor-operators/pkg/controllers/cspc-controller/util"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/klog"
+	"time"
+)
+
+func (c *Controller) UpdateStatus(cspc *cstor.CStorPoolCluster) error {
```

Comment:
Good to change the prototype to 
```go
func (c *Controller) UpdateStatus(cspc *cstor.CStorPoolCluster) (*cstor.CStorPoolCluster, error) {
}
```
So that caller code will get updated CSPC object and next time if someone called the `UpdateStatus` It will succeed in first iteration.

Reply:
How about updating the UpdateStatus to UpdateCSPCStatus? It will be more appropriate IMO :)

Reply:
Here we are in cspc -- we do not update status other than cspc

Reply:
But I have changed the names to look more appropriate